src/main/java/com/textadventure/Lexer.java

```diff
@@ -24,15 +23,13 @@ private TokenType(String pattern) {
 		StringBuffer tokenPatternsBuffer = new StringBuffer();
```

Comment:
Consider changing this to **StringBuilder** instead of **StringBuffer** as it will provide better performance.